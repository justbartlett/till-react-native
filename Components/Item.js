import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import DebitIcon from '../img/debit.svg';
import CreditIcon from '../img/credit.svg';
import {Colors} from '../Theme/Colors';

const Item = ({merchant, amount, type, balance, details}) => {
  return (
    <View style={styles.listItem}>
      <View style={styles.listItemInfo}>
        <View style={styles.smallColumn}>
          {type === 'debit' ? (
            <DebitIcon width={26} height={26} />
          ) : (
            <CreditIcon width={26} height={26} />
          )}
        </View>
        <View style={styles.largeColumn}>
          <Text style={styles.boldText}>{merchant}</Text>
        </View>
        <View style={styles.largeColumn}>
          <Text
            style={[
              styles.textRight,
              type === 'debit' ? styles.redText : styles.greenText,
              styles.boldText,
            ]}>
            {type === 'debit' ? '-' : '+'}
            {amount}
          </Text>
        </View>
        <View style={styles.largeColumn}>
          <Text style={[styles.textRight, styles.boldText]}>${balance}</Text>
        </View>
      </View>
      {details && (
        <View>
          <Text style={[styles.details, styles.semiBoldText]}>{details}</Text>
        </View>
      )}
    </View>
  );
};

export default Item;

const styles = StyleSheet.create({
  semiBoldText: {
    fontFamily: 'NunitoSans-SemiBold',
  },
  details: {
    color: Colors.grey,
    marginLeft: '10%',
  },
  textRight: {
    textAlign: 'right',
  },
  boldText: {
    fontFamily: 'NunitoSans-Bold',
  },
  redText: {
    color: Colors.red,
  },
  greenText: {
    color: Colors.green,
  },
  smallColumn: {
    flex: 1,
  },
  largeColumn: {
    flex: 3,
  },
  listItem: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: Colors.white,
    marginHorizontal: 20,
    marginVertical: 8,
    borderRadius: 8,
    paddingHorizontal: 15,
    paddingVertical: 25,
    shadowColor: Colors.shadow,
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.27,
    shadowRadius: 4.65,
    elevation: 6,
  },
  listItemInfo: {
    flex: 1,
    flexDirection: 'row',
    color: Colors.black,
    fontSize: 14,
    justifyContent: 'space-between',
    alignItems: 'center',
  },
});
