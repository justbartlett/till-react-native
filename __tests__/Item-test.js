/**
 * @format
 */

import 'react-native';
import React from 'react';
import Item from '../Components/Item';
import renderer from 'react-test-renderer';

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<Item />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when where is an item', () => {
  const item = {
    merchant: 'Till, Inc.',
    date: '2019-01-15',
    amount: 1000000,
    type: 'credit',
    details: 'Payroll',
  };
  const tree = renderer
    .create(
      <Item
        merchant={item.merchant}
        amount={item.amount}
        type={item.type}
        details={item.details}
        balance={item.balance}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
