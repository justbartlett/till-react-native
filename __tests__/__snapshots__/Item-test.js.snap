// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "elevation": 6,
      "flex": 1,
      "flexDirection": "column",
      "marginHorizontal": 20,
      "marginVertical": 8,
      "paddingHorizontal": 15,
      "paddingVertical": 25,
      "shadowColor": "#9495CC",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.27,
      "shadowRadius": 4.65,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "color": "#1C1C1C",
        "flex": 1,
        "flexDirection": "row",
        "fontSize": 14,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <SvgMock
        height={26}
        width={26}
      />
    </View>
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "NunitoSans-Bold",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "right",
            },
            Object {
              "color": "#00CE67",
            },
            Object {
              "fontFamily": "NunitoSans-Bold",
            },
          ]
        }
      >
        +
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "right",
            },
            Object {
              "fontFamily": "NunitoSans-Bold",
            },
          ]
        }
      >
        $
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly when where is an item 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "elevation": 6,
      "flex": 1,
      "flexDirection": "column",
      "marginHorizontal": 20,
      "marginVertical": 8,
      "paddingHorizontal": 15,
      "paddingVertical": 25,
      "shadowColor": "#9495CC",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.27,
      "shadowRadius": 4.65,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "color": "#1C1C1C",
        "flex": 1,
        "flexDirection": "row",
        "fontSize": 14,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <SvgMock
        height={26}
        width={26}
      />
    </View>
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "NunitoSans-Bold",
          }
        }
      >
        Till, Inc.
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "right",
            },
            Object {
              "color": "#00CE67",
            },
            Object {
              "fontFamily": "NunitoSans-Bold",
            },
          ]
        }
      >
        +
        1000000
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "right",
            },
            Object {
              "fontFamily": "NunitoSans-Bold",
            },
          ]
        }
      >
        $
      </Text>
    </View>
  </View>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#8A8A8A",
            "marginLeft": "10%",
          },
          Object {
            "fontFamily": "NunitoSans-SemiBold",
          },
        ]
      }
    >
      Payroll
    </Text>
  </View>
</View>
`;
